tfrecorder/schema.py

```diff
@@ -0,0 +1,179 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Defines input types for tfrecorder's input schema. """
+
+import collections
+from typing import Dict, List, Union
+
+import frozendict
+import tensorflow as tf
+import tensorflow_transform as tft
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import schema_utils
+
+# All supported types will be based on _supported_type.
+_supported_type = collections.namedtuple(
+    'tfrecordInputType',
+    ['type_name', 'feature_spec', 'allowed_values'])
+
+# Supported type definitions here.
+image_uri = _supported_type(
+    type_name='image_uri',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+allowed_split_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+split_key = _supported_type(
+    type_name='split_key',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=allowed_split_values)
+
+#TODO(mikebernico): Implement in preprocess_fn
+integer_input = _supported_type(
+    type_name='integer_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.int64),
+    allowed_values=None)
+
+#TODO(mikebernico): Implement in preprocess_fn
+float_input = _supported_type(
+    type_name='float_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.float64),
+    allowed_values=None)
+
+#TODO(mikebernico): Implement in preprocess_fn
+categorical_input = _supported_type(
+    type_name='categorical_input',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+integer_label = _supported_type(
+    type_name='integer_label',
+    feature_spec=tf.io.FixedLenFeature([], tf.int64),
+    allowed_values=None)
+
+string_label = _supported_type(
+    type_name='string_label',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+image_support_type = _supported_type(
+    type_name='image_support_type',
+    feature_spec=tf.io.FixedLenFeature([], tf.string),
+    allowed_values=None)
+
+# Default schema supports the legacy image_csv format.
+image_csv_schema = frozendict.FrozenOrderedDict({
+    'split': split_key,
+    'image_uri': image_uri,
+    'label': string_label})
+
+
+def get_raw_schema_map(
+    schema_map: Dict[str, collections.namedtuple]
+    ) -> Dict[str, collections.namedtuple]:
+  """Converts a schema to a raw (pre TFT / post image extraction) schema."""
+  raw_schema = {}
+  for k, v in schema_map.items():
+    if v.type_name == "image_uri":
+      raw_schema['image_name'] = image_support_type
```

Comment:
`image_support_type` seems a bit vague if it's meant to represent a string.

Reply:
Yeah, I don't think it's correct for the dimensions either.  I changed this to image_support_string and image_support_int